include/drivers/dac.h

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2020 Libre Solar Technologies GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief DAC public API header file.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_DAC_H_
+#define ZEPHYR_INCLUDE_DRIVERS_DAC_H_
+
+#include <device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief DAC driver APIs
+ * @defgroup dac_interface DAC driver APIs
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/** @brief DAC references. */
+enum dac_reference {
+	DAC_REF_DEFAULT,   /**< Default, reference cannot be selected. */
+	DAC_REF_VDD_1,     /**< VDD. */
+	DAC_REF_VDD_1_2,   /**< VDD/2. */
+	DAC_REF_VDD_1_3,   /**< VDD/3. */
+	DAC_REF_VDD_1_4,   /**< VDD/4. */
+	DAC_REF_INTERNAL,  /**< Internal. */
+	DAC_REF_EXTERNAL0, /**< External, input 0. */
+	DAC_REF_EXTERNAL1, /**< External, input 1. */
+};
+
+/**
+ * @brief Structure for specifying the configuration of a DAC channel.
+ */
+struct dac_channel_cfg {
+	/**
+	 * Channel identifier of the DAC that should be configured.
+	 */
+	u8_t channel_id;
+
+	/**
+	 * Reference selection.
+	 */
+	enum dac_reference reference;
+
+	/**
+	 * Enable buffered output if available in underlying hardware.
+	 */
+	bool buffered;
+};
+
+/**
+ * @brief Type definition of DAC API function for configuring a channel.
+ * See dac_channel_setup() for argument descriptions.
+ */
+typedef int (*dac_api_channel_setup)(struct device *dev,
+				     const struct dac_channel_cfg *channel_cfg);
+
+/**
+ * @brief Type definition of DAC API function for setting a write request.
+ * See dac_write_value() for argument descriptions.
+ */
+typedef int (*dac_api_write_value)(struct device *dev,
+			    u8_t channel, u32_t value, u8_t resolution);
```

Comment:
I am not sure if the resolution really belongs in the `write_value()` API call. Perhaps it is better suited in `struct dac_channel_cfg`? Wouldn't it be more logical to configure the resolution once per channel and then just write a value to a channel afterwards?

Reply:
Actually thought the same, but wanted to keep it similar to the ADC. As you marked this as resolved already, I guess we keep it as it is for now?

Reply:
If I marked it as resolved it wasn’t on purpose.

Reply:
Alright, so should I move the resolution setting to channel cfg and break similarity with the ADC? I would be fine with that, but do we want to wait for feedback from others or just go ahead?

Reply:
Let's discuss it on the API meeting Tuesday if you are able to join? I see your point in matching the ADC API, although I would prefer the resolution to be set once there as well.

It's not a cardinal point for me. We can take in the API as-in and then evaluate the resolution parameter as it matures.

Reply:
Makes sense. I should be able to join tomorrow.